test/github-issues/528/issue-528.ts

```diff
@@ -0,0 +1,34 @@
+import "reflect-metadata";
+import {createTestingConnections, closeTestingConnections, reloadTestingDatabases} from "../../utils/test-utils";
+import {Connection} from "../../../src/connection/Connection";
+
+import {Post} from "./entity/Post";
+import {Author} from "./entity/Author";
+
+describe("github issues > #528 Migrations failing on timestamp validation", () => {
+
+    let connections: Connection[];
+    before(async () => connections = await createTestingConnections({
+        entities: [__dirname + "/entity/*{.js,.ts}"],
+        migrations: [__dirname + "/migrations/*{.js,.ts}"],
+        schemaCreate: true,
+        dropSchemaOnConnection: true,
+    }));
+    after(() => closeTestingConnections(connections));
+
```

Comment:
Missing a test for made changes. 
You need to check correct order of undo / apply (at least)